x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/googlevertexai/completion/GoogleVertexAiChatCompletionTaskSettings.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.googlevertexai.completion;
+
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.inference.TaskSettings;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class GoogleVertexAiChatCompletionTaskSettings implements TaskSettings {
+    public static final String NAME = "google_vertex_ai_chatcompletion_task_settings";
+
+    private final ThinkingConfig thinkingConfig;
+
+    public static final GoogleVertexAiChatCompletionTaskSettings EMPTY_SETTINGS = new GoogleVertexAiChatCompletionTaskSettings();
+    private static final ThinkingConfig EMPTY_THINKING_CONFIG = new ThinkingConfig();
+
+    public GoogleVertexAiChatCompletionTaskSettings() {
+        thinkingConfig = EMPTY_THINKING_CONFIG;
+    }
+
+    public GoogleVertexAiChatCompletionTaskSettings(ThinkingConfig thinkingConfig) {
+        this.thinkingConfig = Objects.requireNonNullElse(thinkingConfig, EMPTY_THINKING_CONFIG);
+    }
+
+    public GoogleVertexAiChatCompletionTaskSettings(StreamInput in) throws IOException {
+        if (in.getTransportVersion().onOrAfter(TransportVersions.GEMINI_THINKING_BUDGET_ADDED)) {
```

Comment:
Let's remove the if block and just have `thinkingConfig = new ThinkingConfig(in);`